script.js

```diff
@@ -52,4 +52,112 @@ function valutaConverter(event) {
   }
 }
 
-// updateing existing currency conversion rates with new rates form----------------
+// updateing existing currency conversion rates with new rates form
+// **********************************************
+// Week2 / An aaray of the currency rate objects
+// **********************************************
+
+const brandObjectArray = [
+  {
+    timestamp: Date.now(),
+    base: "EUR",
+    date: new Date().toLocaleDateString(),
+    rates: {},
+  },
+  {
+    timestamp: Date.now(),
+    base: "USD",
+    date: new Date().toLocaleDateString(),
+    rates: {},
+  },
+  {
+    timestamp: Date.now(),
+    base: "DKK",
+    date: new Date().toLocaleDateString(),
+    rates: {},
+  },
+];
+
+document
+  .getElementById("brand-currency-form-beta")
+  .addEventListener("submit", brandCurrencyBeta);
+
+function brandCurrencyBeta(event) {
+  event.preventDefault();
+
+  //getting and filling objects of array------------
+  const baseCurrency = document.getElementById("base-currency-beta").value;
+  const quoteName = document.getElementById("quote-name-beta").value;
+  const quoteRate = document.getElementById("quote-rate-beta").value;
+
+  if (baseCurrency === "EUR") {
+    brandObjectArray[0].rates[quoteName] = +quoteRate;
+  } else if (baseCurrency === "USD") {
+    brandObjectArray[1].rates[quoteName] = +quoteRate;
+  } else if (baseCurrency === "DKK") {
+    brandObjectArray[2].rates[quoteName] = +quoteRate;
+  }
+  document.getElementById("quote-rate-beta").value = "";
+  console.log(brandObjectArray);
+
+  //Calling function: Display all base currencies and their rates
+  displayBaseAndQuoteCurrenciesAndRates();
+  //Calling function: Display currencies in table
+  CurrencyDisplayTable();
+}
+
+//Function: checking base currency and showing existing quotes and rate for each base ---------
+function displayBaseAndQuoteCurrenciesAndRates() {
```

Comment:
as I mentioned above, the user should not see anything in the console, so you can delete it, but next time if you want to loop through the array and log the values out, There is a simpler way

```
  for (currency of brandObjectArray) {
    console.log(`${currency.base} base:`)
    console.log(currency.rates)
  }
```

Reply:
Thank you.